_shared/project/db.py

```diff
@@ -0,0 +1,62 @@
+import alembic.command
+import alembic.config
+{% if cookiecutter.get("_directory") == "pyramid-app" %}
+import zope.sqlalchemy
+{% endif %}
+from sqlalchemy import MetaData{% if cookiecutter.get("_directory") == "pyramid-app" %}, create_engine{% endif %}, text
+from sqlalchemy.exc import ProgrammingError
+from sqlalchemy.orm import DeclarativeBase, MappedAsDataclass{% if cookiecutter.get("_directory") == "pyramid-app" %}, Session{% endif %}
+
+
+
+class Base(MappedAsDataclass, DeclarativeBase):
+    metadata = MetaData(
+        naming_convention={
+            "ix": "ix_%(column_0_N_label)s",
+            "uq": "uq_%(table_name)s_%(column_0_N_name)s",
+            "ck": "ck_%(table_name)s_%(constraint_name)s",
+            "fk": "fk_%(table_name)s_%(column_0_N_name)s_%(referred_table_name)s_%(referred_column_0_N_name)s",
+            "pk": "pk_%(table_name)s",
+        }
+    )
+
+
+def stamp(engine):  # pragma: no cover
+    """If the DB isn't already stamped then stamp it with the latest revision."""
+    stamped = False
+
+    with engine.connect() as connection:
+        try:
+            if connection.execute(text("select * from alembic_version")).first():
+                stamped = True
+        except ProgrammingError:
+            pass
+
+    if not stamped:
+        alembic.command.stamp(alembic.config.Config("conf/alembic.ini"), "head")
```

Comment:
This is a copy-paste of a hack that we already had in LMS and h. It's only called in dev and it just automatically stamps the DB with the latest Alembic revision if it's not already stamped. This just avoids people having to manually stamp their DBs with is a real fiddle